Assets/PatchKit Patcher/Scripts/AppStarter.cs

```diff
@@ -29,11 +29,17 @@ public AppStarter(App app)
 
         private string ResolveExecutablePath(AppVersion appVersion)
         {
+            PlatformType platformType = Platform.GetPlatformType();
+
             if (!string.IsNullOrEmpty(appVersion.MainExecutable))
             {
                 string executablePath = Path.Combine(_app.LocalDirectory.Path, appVersion.MainExecutable);
 
-                if (File.Exists(executablePath))
+                bool isOSXApp = platformType == PlatformType.OSX &&
```

Comment:
Is there any chance that `Platform.GetPlatformType` returns `OSX` but the application isn't an OSX app?

Reply:
Nope, it's always the same.

Reply:
The we shouldn't need all the other conditions, right?

 ```
...
executablePath.EndsWith(".app") &&
Directory.Exists(executablePath);
```

Reply:
This isn't checking whether application platform that secret we are using is OSX, but whether the `appVersion.MainExecutable` points to valid OSX `.app`.